src/hardware/serialport/softmodem.h

```diff
@@ -61,105 +61,102 @@ bool MODEM_ReadPhonebook(const std::string &filename);
 
 class CFifo {
 public:
-	CFifo(Bitu _size)
-		: data(_size),
-		  size(_size),
-		  pos(0),
-		  used(0)
-	{}
-
-	INLINE Bitu left(void) const {
-		return size - used;
-	}
-	INLINE Bitu inuse(void) const {
-		return used;
-	}
-	void clear(void) {
+	CFifo(const size_t n) : data(n), size(n) {}
+
+	uint32_t left() const { return size - used; }
+	uint32_t inuse() const { return used; }
+	void clear()
+	{
 		used = 0;
 		pos = 0;
 	}
 
-	void addb(Bit8u _val) {
-		if(used >= size) {
-			static Bits lcount = 0;
+	void addb(uint8_t val)
+	{
+		if (used >= size) {
+			static uint16_t lcount = 0;
 			if (lcount < 1000) {
 				lcount++;
 				LOG_MSG("MODEM: FIFO Overflow! (addb)");
 			}
 			return;
 		}
 		//assert(used<size);
-		Bitu where = pos + used;
+		size_t where = pos + used;
 		if (where >= size)
 			where -= size;
-		data[where]=_val;
-		//LOG_MSG("+%x",_val);
+		data[where] = val;
+		//LOG_MSG("+%x", val);
 		used++;
 	}
 
-	void adds(Bit8u * _str,Bitu _len) {
-		if ( (used+_len) > size) {
-			static Bits lcount=0;
+	void adds(uint8_t *str, size_t len)
+	{
+		if ((used + len) > size) {
+			static uint16_t lcount = 0;
 			if (lcount < 1000) {
 				lcount++;
-				LOG_MSG("MODEM: FIFO Overflow! (adds len %" PRIuPTR ")",
-				        _len);
+				LOG_MSG("MODEM: FIFO Overflow! (adds len %" PRIu64 ")",
+				        static_cast<uint64_t>(len));
```

Comment:
MSYS2 failed to choke-down the `%zu`, :nauseated_face:
So opted for casting the `size_t` across to a `uint64_t`; which is probably the closest things we've got to a `size_t`, including future-proofing.  MSYS2 is happy with this.

Reply:
Still doesn't like it.  
```
2020-05-18T03:03:52.3865076Z softmodem.h:99:13: warning: unknown conversion type character 'l' in format [-Wformat=]
2020-05-18T03:03:52.3906645Z    99 |     LOG_MSG("MODEM: FIFO Overflow! (adds len %" PRIu64 ")",
2020-05-18T03:03:52.3921072Z       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
I'm going to step-down to `unsigned` and `%u`. Not perfect, but still overkill for whatever value `len` could possibly hold.